internal/pkg/service/common/httpserver/error.go

```diff
@@ -0,0 +1,127 @@
+package httpserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	goaHTTP "goa.design/goa/v3/http"
+	"goa.design/goa/v3/middleware"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/encoding/json"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	. "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	DefaultErrorName    = "internalError"
+	DefaultErrorMessage = "Application error. Please contact our support support@keboola.com with exception id (%s) attached."
+)
+
+type ErrorWriter struct {
+	logger            log.Logger
+	errorNamePrefix   string
+	exceptionIDPrefix string
+}
+
+// FormatError sets HTTP status code to error.
+func FormatError(_ context.Context, err error) goaHTTP.Statuser {
+	return WrapWithStatusCode(err, HttpCodeFrom(err))
+}
+
+func NewErrorWriter(logger log.Logger, errorNamePrefix, exceptionIDPrefix string) ErrorWriter {
+	return ErrorWriter{logger: logger, errorNamePrefix: errorNamePrefix, exceptionIDPrefix: exceptionIDPrefix}
+}
+
+func (wr *ErrorWriter) Write(ctx context.Context, w http.ResponseWriter, err error) {
+	_ = wr.WriteOrErr(ctx, w, err)
+}
+
+func (wr *ErrorWriter) WriteOrErr(ctx context.Context, w http.ResponseWriter, err error) error {
+	// Default values
+	response := &UnexpectedError{
+		StatusCode:  http.StatusInternalServerError,
+		Name:        DefaultErrorName,
+		Message:     err.Error(),
+		ExceptionID: nil,
+	}
+
+	// HTTP status code
+	var errWithStatusCode WithStatusCode
+	if errors.As(err, &errWithStatusCode) {
+		response.StatusCode = errWithStatusCode.StatusCode()
+	}
+
+	// Error name
+	var nameProvider WithName
+	if errors.As(err, &nameProvider) {
+		response.Name = nameProvider.ErrorName()
+	}
+
+	// Normalize error name
+	if !strings.Contains(response.Name, ".") {
+		// Normalize error name, eg., "missing_field" to "buffer.missingField"
+		response.Name = wr.errorNamePrefix + strcase.ToLowerCamel(response.Name)
+	}
+
+	// Re-use exception ID from Storage or other API, if possible.
+	// Otherwise, generate custom exception ID.
+	var exceptionIdProvider WithExceptionId
+	if errors.As(err, &exceptionIdProvider) {
```

Comment:
`exceptionIDProvider` Mohli bysme teda asi zprovoznit ten linter. 😄 Podívám se na to.

Reply:
Tak to neřeš. Jak se zapne ten linter, tak se stejně bude muset pofixovat všude.

Reply:
to fixnem, to je najmenej, kopiroval som to, tak som sa nad tym nezamyslal